File: server/src/models/Store.js
import mongoose from "mongoose";

const storeSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  location: { type: String },
  contact: { type: String },
  foods: [{ type: mongoose.Schema.Types.ObjectId, ref: "FoodListing" }]
}, { timestamps: true });

export default mongoose.model("Store", storeSchema);

File: server/src/models/store.py
from database import db

class Store(db.Model):
    __tablename__ = "stores"

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(120), nullable=False)
    location = db.Column(db.String(255))
    contact = db.Column(db.String(120))

    # relationship: one store has many foods
    foods = db.relationship("FoodListing", back_populates="store")
